### ISSUE TITLE
My Installation Experience with Cuda on Windows

### ISSUE THREAD

**Issue by jjmlovesgit** [resemble-ai/chatterbox#159]
Summary: The Proven Method for Installing Standard resemble-ai/chatterbox with GPU

Phase 1: Environment Preparation
Open Anaconda Prompt.

Create the New Environment:
Bash

conda create -n chatterbox_standard_env python=3.10 -y
Activate the Environment:

Bash
conda activate chatterbox_standard_env

Phase 2: The GPU Foundation
Install the correct, GPU-enabled version of PyTorch using conda. This is the most critical step.
Bash
conda install pytorch torchvision torchaudio pytorch-cuda=12.1 -c pytorch -c nvidia -y

Phase 3: The Manual Transplant of Chatterbox
We will download the official source code and install it, but we will explicitly forbid it from touching our perfect PyTorch installation.
Download the Official Source Code:
Bash
git clone https://github.com/resemble-ai/chatterbox.git
Navigate into the Code Directory:
Bash
cd chatterbox

Install the Code, Ignoring Its Dependencies: The --no-deps flag is the key to our success.
Bash
pip install --no-deps .
(The . means "install the package in the current directory.")

Phase 4: Install All Other Dependencies
Now we install the supporting cast of libraries from your large, corrected requirements.txt file. This file contains everything else your API needs to run.

Prepare the requirements.txt file: Ensure you are using the final, fully corrected version of the file:
*Important (see below for file) - edit provided requirments.txt with torch and chatterbox lines removed, and librosa, tenacity, packaging, and aiohttp versions fixed).
Navigate Back and Install:
Bash
# Go back to your main project directory
cd .. 
# Install all remaining packages
pip install -r requirements.txt

Phase 5: Verification and Usage
After the installation completes, your environment is ready.

Verify the Installation: Run the final check to ensure everything is visible and CUDA is active.
Bash

python -c "import torch; import chatterbox; print('--- STANDARD INSTALL SUCCESS ---'); print(f'PyTorch Version: {torch.__version__}'); print(f'CUDA Available: {torch.cuda.is_available()}')"

Run Your Application: You can now create a Python script to use the standard chatterbox. The usage is identical to the documentation you found earlier:

Python

from chatterbox.tts import ChatterboxTTS

model = ChatterboxTTS.from_pretrained(device="cuda")
wav = model.generate("This is a test.")
# ... etc.

This method ensures you get a stable, GPU-accelerated environment for the official Chatterbox model, completely separate from your streaming 
setup.

_______________________________________

# === Final Corrected Requirements File ===
# torch, torchvision, torchaudio, and chatterbox-tts have been removed
# because they are being installed manually with conda and from git.
# Versions for librosa, tenacity, packaging, and aiohttp have been fixed.

accelerate==1.5.2
aiofiles==23.2.1
aiohttp==3.9.5
aioice==0.10.1
aiortc==1.11.0
altair==5.5.0
annotated-types==0.7.0
antlr4-python3-runtime==4.9.3
anyio==4.9.0
attrs==25.3.0
audioread==3.0.1
av==14.2.0
blinker==1.9.0
cachetools==5.5.2
certifi==2025.1.31
cffi==1.17.1
cfgv==3.4.0
charset-normalizer==3.4.1
click==8.1.8
colorama==0.4.6
coloredlogs==15.0.1
conformer==0.3.2
contourpy==1.3.1
cryptography==44.0.2
cycler==0.12.1
decorator==5.2.1
diffusers==0.29.0
distlib==0.3.9
dnspython==2.7.0
einops==0.8.1
fastapi==0.115.11
ffmpeg-python==0.2.0
ffmpy==0.5.0
filelock==3.13.1
flatbuffers==25.2.10
fonttools==4.56.0
fsspec==2024.6.1
future==1.0.0
gitdb==4.0.12
GitPython==3.1.44
google-crc32c==1.7.1
gradio==5.23.3
gradio_client==1.8.0
groovy==0.1.2
h11==0.14.0
httpcore==1.0.7
httpx==0.28.1
huggingface-hub==0.29.3
humanfriendly==10.0
identify==2.6.12
idna==3.10
ifaddr==0.2.0
imageio==2.37.0
importlib_metadata==8.6.1
importlib_resources==6.5.2
Jinja2==3.1.4
joblib==1.4.2
jsonschema==4.23.0
jsonschema-specifications==2024.10.1
kiwisolver==1.4.8
lazy_loader==0.4
librosa==0.11.0
llvmlite==0.44.0
markdown-it-py==3.0.0
MarkupSafe==2.1.5
matplotlib==3.10.1
mdurl==0.1.2
more-itertools==10.6.0
mpmath==1.3.0
msgpack==1.1.0
narwhals==1.32.0
networkx==3.3
nltk==3.9.1
nodeenv==1.9.1
numba==0.61.0
numpy==1.26.0
omegaconf==2.3.0
onnx==1.18.0
onnxruntime-gpu==1.21.1
openai-whisper==20240930
opencv-python==4.11.0.86
orjson==3.10.15
packaging==23.2
pandas==2.2.3
peft==0.15.0
pillow==11.0.0
platformdirs==4.3.7
pooch==1.8.2
pre_commit==4.2.0
protobuf==5.29.4
psutil==5.9.8
pyarrow==19.0.1
pycparser==2.22
pydantic==2.10.6
pydantic_core==2.27.2
pydeck==0.9.1
pydub==0.25.1
pyee==13.0.0
Pygments==2.19.1
pylibsrtp==0.12.0
pyOpenSSL==25.0.0
pyparsing==3.2.1
pyreadline3==3.5.4
pysbd==0.3.4
python-dateutil==2.9.0.post0
python-dotenv==1.0.1
python-multipart==0.0.20
pytz==2025.1
PyYAML==6.0.2
qwen-vl-utils==0.0.2
referencing==0.36.2
regex==2024.11.6
resampy==0.4.3
resemble-perth==1.0.1
rich==13.9.4
rpds-py==0.23.1
ruff==0.11.2
s3tokenizer==0.1.7
safehttpx==0.1.6
safetensors==0.5.3
scikit-learn==1.6.1
scipy==1.15.2
semantic-version==2.10.0
sentencepiece==0.2.0
setuptools==70.2.0
shellingham==1.5.4
simli-ai==0.1.11
six==1.17.0
smmap==5.0.2
snac==1.2.1
sniffio==1.3.1
soundfile==0.13.1
soxr==0.5.0.post1
spaces==0.32.0
spicy==0.16.0
starlette==0.46.1
streamlit==1.44.0
sympy==1.13.1
tenacity==8.2.2
threadpoolctl==3.6.0
tiktoken==0.9.0
tokenizers==0.20.3
toml==0.10.2
tomlkit==0.12.0
tornado==6.4.2
tqdm==4.67.1
transformers==4.46.3
transformers-stream-generator==0.0.4
typer==0.15.2
typing_extensions==4.12.2
tzdata==2025.1
urllib3==2.3.0
uvicorn==0.34.0
virtualenv==20.31.2
watchdog==6.0.0
Wave==0.0.2
websockets==15.0.1
zipp==3.21.0

**Comment by jjmlovesgit** [comment]
Why do we need this?


The packages on PyPI (chatterbox-tts and chatterbox-streaming) were published with a setup file that explicitly demanded torch==2.6.0.

The torch==2.6.0 available for your Python version was CPU-only.

**Your RTX 3090 requires a specific GPU-enabled version of PyTorch (like the pytorch-2.5.1-py3.10_cuda12.1 we installed with conda).**

These two requirements are mutually exclusive. You cannot have both. Any attempt to let pip install the package automatically would cause it to rip out our working GPU PyTorch and install the CPU version, breaking everything.

Our final "surgical" plan succeeded because it directly addressed every single one of these root causes:

We used conda: 
We installed PyTorch First: We manually installed the correct, non-negotiable GPU foundation before anything else could interfere.
We Installed from Source with --no-deps: We downloaded the Chatterbox code directly and used the --no-deps flag. This was the key that allowed us to use the application's code while completely ignoring its broken, hardcoded dependency list.
We Manually Curated the Final requirements.txt: We acted as the dependency resolver ourselves, fixing the version conflicts for librosa, tenacity, etc., creating a final list that was guaranteed to be compatible with our foundation.

**Comment by allanrbo** [comment]
Can be done simpler actually. Here's what I did on my Debian 12 box, for which `nvidia-driver/stable,now 535.247.01-1~deb12u1 amd64` is the most up to date as of writing.

```
mkdir chatterboxtest
cd chatterboxtest
python3 -m venv venv
. venv/bin/activate

cat > requirements.txt <<"EOF"
--extra-index-url https://download.pytorch.org/whl/cu121
torch==2.6.0
torchaudio==2.6.0
EOF
pip install -r requirements.txt

# Verify that torch with cuda support works at this point. This should end with "True".
python -c "import torch; print(torch.cuda.is_available())"

git clone https://github.com/resemble-ai/chatterbox.git
cd chatterbox
pip install .
# Edit the pyproject.toml to remove the deps for torch and torchaudio, because we have already installed them in our venv.
sed -i '/"torch==/Id; /"torchaudio==/Id' pyproject.toml
pip install .
cd ..

cat > gen.py <<"EOF"
import torchaudio as ta
from chatterbox.tts import ChatterboxTTS
model = ChatterboxTTS.from_pretrained(device="cuda")
wav = model.generate("Hello world! How's it going.")
ta.save("test-1.wav", wav, model.sr)
EOF
python3 gen.py
```
